src/blocks/analysis.rs

```diff
@@ -78,7 +75,6 @@ where
                     } => {
                         let n: usize = input_chunk.len();
                         if Some(n) != previous_chunk_len {
-                            fft = Some(FftPlanner::<Flt>::new().plan_fft_forward(n));
```

Comment:
Here you create a new planner on each iteration of the loop right?

Reply:
No, only for the first iteration of the loop (unless the chunk size changes). See this if-clause:

```rust
if Some(n) != previous_chunk_len {
    fft = Some(FftPlanner::<Flt>::new().plan_fft_forward(n));
```

([source](https://github.com/JanBeh/radiorust/blob/0.5.0/src/blocks/analysis.rs#L80))

If the planner was generated for every iteration of the loop, I would consider it a bug. But it's only (re-)generated when the chunk-size changes.

Reply:
Ah, I see it only happens if the size of the chunk buffer changes.